misc.go

```diff
@@ -53,6 +53,19 @@ func runeSliceWidthRange(rs []rune, beg, end int) []rune {
 	return rs[b:]
 }
 
+// Returns the last runes of `rs` that take up at most `max_width` space.
+func runeSliceWidthLastRange(rs []rune, max_width int) []rune {
+	last_width := 0
```

Comment:
Regarding naming conventions, the recommendation for Go is to use camelcase, see https://go.dev/doc/effective_go#mixed-caps

```suggestion
	lastWidth := 0
```